src/test/java/de/azapps/kafkabackup/common/record/RecordJSONSerdeTest.java

```diff
@@ -25,67 +29,136 @@
+                "]";
+    }
+
+    private byte[] jsonWithAllFields() {
+        return ("{"+
+                    String.format("\"topic\":\"%s\",", topic)+
+                    String.format("\"partition\":%d,", partition)+
+                    String.format("\"offset\":%d,", offset)+
+                    String.format("\"timestamp_type\":\"%s\",", timestampType)+
+                    String.format("\"timestamp\":%d,", timestamp)+
+                    String.format("\"key\":\"%s\",", keyBase64)+
+                    String.format("\"value\":\"%s\",", valueBase64)+
+                    String.format("\"headers\":%s", jsonHeadersStr())+
+                "}").getBytes(JSON_ENCODING);
+    }
+
+    private byte[] jsonWithNullKeyAndValue() {
+        return ("{"+
+                String.format("\"topic\":\"%s\",", topic)+
+                String.format("\"partition\":%d,", partition)+
+                String.format("\"offset\":%d,", offset)+
+                String.format("\"timestamp_type\":\"%s\",", timestampType)+
+                String.format("\"timestamp\":%d,", timestamp)+
+                String.format("\"key\":%s,", "null")+
+                String.format("\"value\":%s,", "null")+
+                String.format("\"headers\":%s", jsonHeadersStr())+
+                "}").getBytes(JSON_ENCODING);
+    }
+
+    private byte[] jsonWithNoTimestampType() {
+        return ("{"+
+                String.format("\"topic\":\"%s\",", topic)+
+                String.format("\"partition\":%d,", partition)+
+                String.format("\"offset\":%d,", offset)+
+                String.format("\"timestamp_type\":\"%s\",", TimestampType.NO_TIMESTAMP_TYPE)+
+                String.format("\"timestamp\":%s,", "null")+
+                String.format("\"key\":\"%s\",", keyBase64)+
+                String.format("\"value\":\"%s\",", valueBase64)+
+                String.format("\"headers\":%s", jsonHeadersStr())+
+                "}").getBytes(JSON_ENCODING);
     }
 
-    private byte[] jsonWithNullKeyAndValue() throws UnsupportedEncodingException {
-        return String.format("{\"topic\":\"%s\",\"partition\":%d,\"offset\":%d,\"timestamp_type\":\"%s\",\"timestamp\":%d,\"key\":null,\"value\":null}", topic, partition, offset, timestampType, timestamp).getBytes(JSON_ENCODING);
+    private byte[] jsonWithEmptyHeaders() {
+        return ("{"+
+                String.format("\"topic\":\"%s\",", topic)+
+                String.format("\"partition\":%d,", partition)+
+                String.format("\"offset\":%d,", offset)+
+                String.format("\"timestamp_type\":\"%s\",", timestampType)+
+                String.format("\"timestamp\":%d,", timestamp)+
+                String.format("\"key\":\"%s\",", keyBase64)+
+                String.format("\"value\":\"%s\",", valueBase64)+
+                String.format("\"headers\":%s", "[]")+
+                "}").getBytes(JSON_ENCODING);
     }
```

Comment:
let's maybe move this to separated file?

Reply:
Yes. Will do in a follow-up PR